libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts

```diff
@@ -463,13 +508,78 @@ describe('Text editor', () => {
       validateToolbarActions(testComponent.toolbarActions);
     });
 
+    [
+      {
+        desc: 'new window',
+        windowOption: 'new',
+        urlStrings: [
+          'href="https://google.com',
+          'rel="noopener noreferrer"',
+          'target="_blank"',
+        ],
+        containsHintTextExpect: (
+          inputBoxText = document.querySelector('.sky-modal .sky-input-box')
+            ?.textContent
+        ) =>
+          expect(inputBoxText).toContain(
+            'This link will open in a new window.'
+          ),
+      },
+      {
+        desc: 'current window',
+        windowOption: 'existing',
+        urlStrings: ['<a', 'href="https://google.com">'],
+        containsHintTextExpect: (
+          inputBoxText = document.querySelector('.sky-modal .sky-input-box')
+            ?.textContent
+        ) =>
+          expect(inputBoxText).not.toContain(
+            'This link will open in a new window.'
+          ),
+      },
+    ].forEach((testArgs) => {
+      it(`should show correct link window options when ${testArgs.desc} option is specified`, fakeAsync(() => {
+        testComponent.value = '<p>Click here</p>';
+        if (testArgs.windowOption === 'new') {
+          testComponent.linkWindowOptions = ['new'];
+          console.log('josie in new');
```

Comment:
Did you mean to remove this after debugging?

Reply:
Oops, removed.